frontend/src/features/userProfile/state/userProfileActions.ts

```diff
@@ -0,0 +1,78 @@
+import { ApiAction, ApiActionThunk, CALL_API } from "../../../store/api";
+
+import { NewSearchResult } from '../../../qri/search'
+import { UserProfileParams } from '../../../qri/userProfile'
+
+export const searchPageSizeDefault = 25
+
+export function loadUserProfile(username: string): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch(fetchUserProfile(username))
+  }
+}
+
+export function loadUserProfileDatasets(username: string, userProfileParams: UserProfileParams): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch(fetchUserProfileDatasets(username, userProfileParams))
+  }
+}
+
+export function loadUserProfileFollowing(username: string, userProfileParams: UserProfileParams): ApiActionThunk {
+  return async (dispatch, getState) => {
+    return dispatch(fetchUserProfileFollowing(username, userProfileParams))
+  }
+}
+
+function fetchUserProfile (username: string): ApiAction {
+  return {
+    type: 'userprofile',
+    [CALL_API]: {
+      endpoint: `identity/profile/${username}`,
+      method: 'GET',
+    }
+  }
+}
+
+const mapSearchResults = (results: any[]) => {
+  return results.map((d) => NewSearchResult(d))
+}
+
+const mapFrontendParams = (frontendParams: UserProfileParams) => {
+  // map frontend 'sort' param to backend 'orderBy'
+  return {
+    orderBy: frontendParams.sort === 'recentlyupdated' ? 'created,desc' : 'name,asc'
+  }
+}
```

Comment:
just a thought, maybe moving the mapping functions from the `actions` file and `reducer` file to a separate `map` file & we know that we are done syncing between the two systems when we can remove the functions that aren't just a one to one conversion to the given interface?

Reply:
I think I prefer them nearby where they are used so they're in your face if you visit this file.  If we end up with a lot of them or lots of duplication, we could centralize.